### DISCUSSION TITLE
lotus client query-ask fault

### DISCUSSION THREAD

**Discussion by lbj2004032** [filecoin-project/lotus#9951] [General Q&A]
$ lotus-miner net reachability
AutoNAT status:  Public
Public address:  /ip4/.../tcp/...

$ lotus client query-ask f0...
ERROR: failed to open stream to miner: exhausted 15 attempts but failed to open stream, err: failed to dial 12D3KooWG4FhdcviYjByYhpaxDQ8Mq4B5UUMWyGDd1zaVwXcmHjx:
  * [/ip4/.../tcp/...] failed to negotiate security protocol: peer id mismatch: expected 12D3KooWG4FhdcviYjByYhpaxDQ8Mq4B5UUMWyGDd1zaVwXcmHjx, but remote key matches 12D3KooWLC5U93nYGDTQipYAYPKp8JK3zUAGmAxqiyUqCBU7UnPG

**Comment by Reiers** [comment]
This is the correct peerID as stated in the Error message:
`12D3KooWLC5U93nYGDTQipYAYPKp8JK3zUAGmAxqiyUqCBU7UnPG`

How to check the ID:
`lotus net id f0..`

**Reply by lbj2004032** [reply]
lotus-miner actor set-peer-id ?

**Comment by beck-8** [comment]
The environment is wrong. Check it carefully.
